xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/NotificationFilterPreferenceManager.java

```diff
@@ -118,6 +119,22 @@ Stream<NotificationFilterPreference> getFilterPreferences(
      */
     void deleteFilterPreference(DocumentReference user, String filterPreferenceId) throws NotificationException;
 
+    /**
+     * Delete a set of filter preferences.
+     * @param user the user to use
+     * @param filterPreferenceIds ids of the filter preferences
+     * @throws NotificationException if an error happens
+     *
+     * @since 15.10.2
+     * @since 16.0.0RC1
+     */
+    @Unstable
+    default void deleteFilterPreferences(DocumentReference user, Set<String> filterPreferenceIds)
```

Comment:
Wouldn't `Collection<String>` be enough for the `filterPreferenceIds` parameter?

Reply:
Hmmm I prefer that the API shows that there shouldn't be any duplicate of ids there: I doubt it would be really an issue, but frankly not entirely sure how a `IN ()` clause for a delete would handle duplicate ids.